src/stcal/saturation/saturation.py

```diff
@@ -115,6 +101,25 @@ def flag_saturated_pixels(
             del flagarray
             del flaglowarray
 
+            if read_pattern is not None:
+                if group == 2:
+                    # Identify groups which we wouldn't expect to saturate by the third group,
+                    # on the basis of the first group
+                    mask = data[ints, 0, ...] / np.mean(read_pattern[0]) * read_pattern[2][-1] < sat_thresh
+
+                    # Identify groups with suspiciously large values in the second group
+                    mask &= data[ints, 1, ...] > sat_thresh / len(read_pattern[1])
+
+                    # Identify groups that are saturated in the third group
+                    mask &= np.where(gdq[ints, 2, :, :] & saturated, True, False)
+
+                    # now, flag any pixels that border saturated pixels
+                    if n_pix_grow_sat > 0:
+                        mask = adjacent_pixels(mask, saturated, n_pix_grow_sat)
```

Comment:
I don't think we can call adjacent_pixels directly; it takes an array with dq flag values, and mask is an array of True/False.

Reply:
yes, I thought I changed this but now it should be ok. @drlaw1558 can you please verify?